src/common.jl

```diff
@@ -72,6 +72,33 @@ nsamples(::Type{D}, x::AbstractMatrix) where {D<:Sampleable{Multivariate}} = siz
 nsamples(::Type{D}, x::Number) where {D<:Sampleable{Matrixvariate}} = 1
 nsamples(::Type{D}, x::Array{Matrix{T}}) where {D<:Sampleable{Matrixvariate},T<:Number} = length(x)
 
+function Base.:(==)(s1::A, s2::B) where {A<:Sampleable, B<:Sampleable}
+    nameof(A) == nameof(B) || return false
+    fields = fieldnames(A)
+    fields == fieldnames(B) || return false
+
+    for f in fields
+        isdefined(s1, f) && isdefined(s2, f) || return false
+        getfield(s1, f) == getfield(s2, f) || return false
+    end
+
+    return true
+end
+
+function Base.isequal(s1::A, s2::B) where {A<:Sampleable, B<:Sampleable}
+    nameof(A) == nameof(B) || return false
+    fields = fieldnames(A)
+    fields == fieldnames(B) || return false
+
+    for f in fields
+        isdefined(s1, f) && isdefined(s2, f) || return false
+        isequal(getfield(s1, f), getfield(s2, f)) || return false
+    end
+
+    return true
+end
```

Comment:
rather than maintain this somewhat complicated code in two places, we could write it like
```julia
for func in (:(==), :isequal)
    @eval function Base.$func(s1::A, s2::B) where {A<:Sampleable, B<:Sampleable}
        nameof(A) == nameof(B) || return false
        fields = fieldnames(A)
        fields == fieldnames(B) || return false

        for f in fields
            isdefined(s1, f) && isdefined(s2, f) || return false
            isequal(getfield(s1, f), getfield(s2, f)) || return false
        end

        return true
    end
end
```	
to my eyes that's clearer that it's the same logic in both cases

Reply:
also defining `isapprox` would then just be adding it to the list of funs we loop over (and change the signature to allow `kwargs`)